python/lsst/afw/math/spatialCell.cc

```diff
@@ -21,16 +21,218 @@
-//#include <pybind11/stl.h>
+#include <pybind11/operators.h>
+#include <pybind11/stl.h>
+
+#include "lsst/afw/geom/Box.h"
+
+#include "lsst/afw/math/SpatialCell.h"
 
 namespace py = pybind11;
+using namespace pybind11::literals;
 
 using namespace lsst::afw::math;
 
+PYBIND11_DECLARE_HOLDER_TYPE(MyType, std::shared_ptr<MyType>);
+
+void declareTestClasses(py::module &mod){
+    /*
+     * Test class for SpatialCellCandidate
+     */
+    class TestCandidate : public lsst::afw::math::SpatialCellCandidate {
+    public:
+        TestCandidate(float const xCenter, ///< The object's column-centre
+                      float const yCenter, ///< The object's row-centre
+                      float const flux     ///< The object's flux
+                    ) :
+            SpatialCellCandidate(xCenter, yCenter), _flux(flux) {
+        }
+
+        /// Return candidates rating
+        virtual double getCandidateRating() const {
+            return _flux;
+        }
+        virtual void setCandidateRating(double flux) {
+            _flux = flux;
+        }
+    private:
+        double _flux;
+    };
+
+
+    /// A class to pass around to all our TestCandidates
+    class TestCandidateVisitor : public lsst::afw::math::CandidateVisitor {
+    public:
+        TestCandidateVisitor() : lsst::afw::math::CandidateVisitor(), _n(0) {}
+
+        // Called by SpatialCellSet::visitCandidates before visiting any Candidates
+        void reset() { _n = 0; }
+
+        // Called by SpatialCellSet::visitCandidates for each Candidate
+        void processCandidate(lsst::afw::math::SpatialCellCandidate *candidate) {
+            ++_n;
+        }
+
+        int getN() const { return _n; }
+    private:
+        int _n;                         // number of TestCandidates
+    };
+    
+    
+/*
+     * Test class for SpatialCellMaskedImageCandidate
+     */
+    class TestMaskedImageCandidate : public lsst::afw::math::SpatialCellMaskedImageCandidate<float> {
+    public:
+        typedef lsst::afw::image::MaskedImage<float> MaskedImageT;
+
+        TestMaskedImageCandidate(float const xCenter, ///< The object's column-centre
+                                 float const yCenter, ///< The object's row-centre
+                                 float const flux     ///< The object's flux
+                                ) :
+            lsst::afw::math::SpatialCellMaskedImageCandidate<float>(xCenter, yCenter), _flux(flux) {
+        }
+
+        /// Return candidates rating
+        double getCandidateRating() const {
+            return _flux;
+        }
+
+        /// Return the %image
+        MaskedImageT::ConstPtr getMaskedImage() const {
+            if (_image.get() == NULL) {
+                _image = MaskedImageT::Ptr(new MaskedImageT(lsst::afw::geom::ExtentI(getWidth(), getHeight())));
+                *_image->getImage() = _flux;
+            }
+
+            return _image;
+        }
+    private:
+        double _flux;
+    };
```

Comment:
In the SWIG version there was a tests/testLib.i file where there classes are defined, so I just added them here. Where would you prefer me to add them?